src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+    this.validateEndpointIpAddressRange(deploymentType, ontapConfiguration.endpointIpAddressRange);
+    this.validateFsxAdminPassword(ontapConfiguration.fsxAdminPassword);
+    this.validateSubnets(deploymentType, props.vpcSubnets, ontapConfiguration.preferredSubnet);
+    this.validateRouteTables(deploymentType, ontapConfiguration.routeTables);
+    this.validateThroughputCapacity(
+      deploymentType,
+      ontapConfiguration.throughputCapacity,
+      ontapConfiguration.throughputCapacityPerHaPair,
+      ontapConfiguration.haPairs,
+    );
+    this.validateStorageCapacity(ontapConfiguration.haPairs, props.storageCapacityGiB);
+  }
+
+  private validateAutomaticBackupRetention(automaticBackupRetention?: Duration): void {
+    if (
+      automaticBackupRetention == null ||
+      automaticBackupRetention.isUnresolved() ||
+      automaticBackupRetention.toMilliseconds() === 0
+    ) {
+      return;
+    }
+    if (
+      automaticBackupRetention.toMilliseconds() < Duration.days(1).toMilliseconds() ||
+      automaticBackupRetention.toDays() > 90
+    ) {
+      throw new Error('automaticBackupRetention must be between 1 and 90 days or be equal to 0');
+    }
+  }
+
+  private validateDailyAutomaticBackupStartTime(
+    automaticBackupRetention?: Duration,
+    dailyAutomaticBackupStartTime?: aws_fsx.DailyAutomaticBackupStartTime,
+  ): void {
+    if (!dailyAutomaticBackupStartTime) {
+      return;
+    }
+
+    const automaticBackupDisabled = !automaticBackupRetention || automaticBackupRetention?.toDays() === 0;
+
+    if (dailyAutomaticBackupStartTime && automaticBackupDisabled) {
+      throw new Error(
+        "'automaticBackupRetention' period must be set a non-zero day when 'dailyAutomaticBackupStartTime' is set",
+      );
+    }
+  }
+
+  private validateDiskIops(storageCapacityGiB: number, diskIops?: number, haPairs: number = 1): void {
+    if (
+      diskIops == null ||
+      Token.isUnresolved(diskIops) ||
+      Token.isUnresolved(storageCapacityGiB) ||
+      Token.isUnresolved(haPairs)
+    ) {
+      return;
+    }
+
+    const minDiskIops = storageCapacityGiB * 3 * haPairs;
+    const maxDiskIops = 200_000 * haPairs;
+
+    if (!Number.isInteger(diskIops) || diskIops < minDiskIops || diskIops > maxDiskIops) {
+      throw new Error(`\'diskIops\' must be an integer between ${minDiskIops} and ${maxDiskIops}, got ${diskIops}`);
+    }
+  }
+
+  private validateEndpointIpAddressRange(deploymentType: OntapDeploymentType, endpointIpAddressRange?: string): void {
+    if (endpointIpAddressRange == null || Token.isUnresolved(endpointIpAddressRange)) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'endpointIpAddressRange' can only be specified for multi-AZ file systems");
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{9,17}$/.test(endpointIpAddressRange)) {
```

Comment:
This regex patter is described in cfn docs.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-ontapconfiguration.html#cfn-fsx-filesystem-ontapconfiguration-endpointipaddressrange